### PR TITLE
easyopts: Deprecate _ALLLOW and replace with _ALLOW

### PR DESCRIPTION
**Pull request by danielgustafsson**
Silently accept CURLOPT_MAIL_RCPT_ALLOWFAILS as an alias for the typoed original version CURLOPT_MAIL_RCPT_ALLLOWFAILS for those who spell it right in their code without even realizing we got it wrong.

### CODE REVIEW

**Comment by jay** [comment]
There's a section in curl.h for aliases, I think that would be better.

https://github.com/curl/curl/blob/cd95ee9f771361acf241629d2fe5507e308082a2/include/curl/curl.h#L691-L699

**Comment by danielgustafsson** [comment]
> On 7 Dec 2022, at 00:27, Jay Satiro ***@***.***> wrote:
> 
> ﻿
> There's a section in curl.h for aliases, I think that would be better.
> 
> https://github.com/curl/curl/blob/cd95ee9f771361acf241629d2fe5507e308082a2/include/curl/curl.h#L691-L699
> 
Oh, nice! I had missed that, will fix up the PR

**Comment by danielgustafsson** [comment]
Pushed a rebase which now use the established aliasing feature for OPTs.

**Comment by danielgustafsson** [comment]
Hmm .. the manpage-syntax test identifies this as an error since it's adding an alias to a non-deprecated option. Not sure what the best way forward is, deprecating ALLLOW and replacing it all with ALLOW? (that would make it a 7.88 PR instead of 7.87 of course)

**Comment by bagder** [comment]
Yeah, I think it makes sense to properly deprecate the wrongly spelled version.

**Comment by danielgustafsson** [comment]
> Yeah, I think it makes sense to properly deprecate the wrongly spelled version.

Thinking more about it I agree. I'll go update the PR to see what that would look like.

File: include/curl/curl.h [hunk 1]
```diff
@@ -2236,7 +2239,6 @@ typedef enum {
 #undef CURLOPT_DNS_USE_GLOBAL_CACHE /* soon obsolete */
 #endif
 
-
```
Comment (danielgustafsson) [inline include/curl/curl.h:2239]: I missed that this spurious whitespace hunk made it in, I'll remove.

**Comment by danielgustafsson** [comment]
This version is a few bricks shy of a load, will have a fixup coming.

**Comment by bagder** [comment]
@danielgustafsson any progress on this?

**Comment by danielgustafsson** [comment]
It admittedly slipped my mind during the holiday season, I'll fix it up.

**Comment by danielgustafsson** [comment]
> @danielgustafsson any progress on this?

I think this version is ready for review.

File: include/curl/curl.h [hunk 2]
```diff
@@ -2203,6 +2204,9 @@ typedef enum {
   /* Can leak things, gonna exit() soon */
   CURLOPT(CURLOPT_QUICK_EXIT, CURLOPTTYPE_LONG, 322),
 
+  /* allow RCPT TO command to fail for some recipients */
+  CURLOPT(CURLOPT_MAIL_RCPT_ALLOWFAILS, CURLOPTTYPE_LONG, 323),
+
```
[the remark this answers is not in the document]
Comment (danielgustafsson) [reply include/curl/curl.h:2209]: Correct me if I'm wrong, but if we do can't deprecate the misspelled one but will be dragging that along? If we do deprecate it with "just" an alias define then all sorts of tests checking for deprecation status fail (my memory is rusty but I did this initially and 1222, 1276 and a few more failed I think).

[the remark this answers is not in the document]
Comment (danielgustafsson) [reply include/curl/curl.h:2209]: > Correct me if I'm wrong, but if we do then we can't deprecate the misspelled one but will be dragging that along? If we do actually deprecate it with "just" an alias define as fallback/implementation then all sorts of tests checking for deprecation status fail (my memory is rusty but I did this initially and 1222, 1276 and a few more failed I think).

**Pull request closed without merging**